### PR TITLE
feat: support SameSite=None

### PR DESCRIPTION
**Pull request by ziyunfei**
close eggjs/egg#3747

### CODE REVIEW

File: lib/cookie.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ class Cookie {
     this.attrs = mergeDefaultAttrs(attrs);
     assert(!this.attrs.path || fieldContentRegExp.test(this.attrs.path), 'argument option path is invalid');
     assert(!this.attrs.domain || fieldContentRegExp.test(this.attrs.domain), 'argument option domain is invalid');
-    assert(!this.attrs.sameSite || this.attrs.sameSite === true || sameSiteRegExp.test(this.attrs.sameSite), 'argument option sameSite is invalid');
+    assert([ undefined, true, false ].indexOf(this.attrs.sameSite) > -1 || sameSiteRegExp.test(this.attrs.sameSite), 'argument option sameSite is invalid');
```
Comment (fengmk2) [inline lib/cookie.js]: need to support `null` value, as same as `undefined`.

File: lib/cookie.js [hunk 2]
```diff
@@ -49,6 +50,7 @@ class Cookie {
     if (attrs.expires) header += '; expires=' + attrs.expires.toUTCString();
     if (attrs.domain) header += '; domain=' + attrs.domain;
     if (attrs.sameSite) header += '; samesite=' + (attrs.sameSite === true ? 'strict' : attrs.sameSite.toLowerCase());
+    if (attrs.sameSite === false) header += '; samesite=none';
```
Comment (fengmk2) [inline lib/cookie.js]: this will breaking change, we should keep false and set nothing.

Comment (ziyunfei) [reply lib/cookie.js]: If any breaking change is unacceptable, then all falsy value (false, null, undefined, 0, '', NaN)  should be treated as a no-op.

**Comment by atian25** [comment]
ci fail

File: lib/cookie.js (hunk 1 above)
Comment (anthonyli) [reply lib/cookie.js]: @fengmk2 更新最新的版本发现没代码，是还没有发包？

**Pull request merged**